api/src/main/java/org/geysermc/geyser/api/item/custom/CustomItemData.java

```diff
@@ -114,6 +115,83 @@ public interface CustomItemData {
      */
     @NonNull Set<String> tags();
 
+    /**
+     * Gets the stack size of the item.
+     *
+     * Returns 0 if not set. When not set (or 0), it defaults to the stack count of the Java item when based of a vanilla item, or 64 when registering a non-vanilla item.
+     *
+     * @return the stack size of the item
+     */
+    @NonNegative
+    int stackSize();
+
+    /**
+     * Gets the max damage of the item.
+     *
+     * Returns -1 if not set. When not set (or below 0), it defaults to the maximum damage of the Java item when based of a vanilla item, or uses 0 when registering a non-vanilla item.
+     *
+     * @return the max damage of the item
+     */
+    int maxDamage();
+
+    /**
+     * Gets the attack damage of the item.
+     * This is purely visual, and only applied to tools
+     *
+     * Returns 0 if not set. When 0, takes the Java item attack damage when based of a vanilla item, or uses 0 when porting a modded item.
+     *
+     * @return the attack damage of the item
+     */
+    int attackDamage();
+
+    /**
+     * Gets the armor type of the item.
+     *
+     * This can be "boots", "leggings", "chestplate", or "helmet", and makes the item able to be equipped into its respective equipment slot.
+     * This should only be set if the Java vanilla/non-vanilla item is able to fit into the specified equipment slot.
```

Comment:
```suggestion
     * This should only be set if the Java item can be placed into the specified equipment slot.
```
If it applies to both, no need to mention it twice